### ISSUE TITLE
class MetricsOutput function select(metrics_selector) bug 

### ISSUE THREAD

**Issue by ds-dba** [capitalone/cloud-custodian#2871]
Hello,

We found the following issue after updating our cloud custodian code to the latest release: 
```
File "/var/task/c7n/output.py", line 60, in select
if k.startswith(metrics_selector):
TypeError: startswith first arg must be str, unicode, or tuple, not bool 
```

This if statement is defined in the select function in the c7n/output.py . 
This function is called in the ctx.py: 
```
metrics_enabled = getattr(options, 'metrics_enabled', None)
factory = MetricsOutput.select(metrics_enabled)
```

We define metrics_enabled in our yml file the following way: 
```
      execution-options:
          metrics_enabled: true
```
This is the same as how it is documented in docs/source/policy/lambda.rst 

as you can see this is called with a boolean variable, however, the startswith expects a string. Could you please look into this? Previously this was done the following way: 
```
    @staticmethod
    def select(metrics_enabled):
        if metrics_enabled:
            return MetricsOutput
        return NullMetricsOutput
```

Thank you in advance. 

Regards, Peter

**Comment by kapilt** [comment]
yeah the latest release started to support metrics as a string to enable pushing metrics across cross providers. we handled the cli compatibility, but execution-options was missing. This also makes me wonder though that we're not defaulting to metrics output on the lambdas, which actually was an intent/goal for lambda policies, ie this explicit config shouldn't be needed, though it should be supported. i'll have a look at both, thanks for the issue report.

**Comment by kapilt** [comment]
merging a fix shortly will release 0.8.31.1 with it in the next 24 hrs.

**Comment by ds-dba** [comment]
thank you  for the quick response!
